qiime2/core/format.py

```diff
@@ -19,15 +19,19 @@ def __init__(self, path=None, mode='w'):
             if mode != 'r':
                 raise ValueError("A path must be omitted when writing.")
 
+        self._mode = mode
```

Comment:
DBC: Can you explain why this move was necessary, it isn't immediately clear to me? Perhaps putting a comment inline would help, too.

Reply:
I added a comment in the new commit. Basically with `self._mode = mode` down at the bottom, we would get validation errors due to objects not have a `._mode` field.